src/hotspot/share/ci/ciTypeFlow.cpp

```diff
@@ -404,11 +404,8 @@ const ciTypeFlow::StateVector* ciTypeFlow::get_start_state() {
     state->push_translate(str.type());
   }
   // Set the rest of the locals to bottom.
-  Cell cell = state->next_cell(state->tos());
-  state->set_stack_size(0);
-  int limit = state->limit_cell();
-  for (; cell < limit; cell = state->next_cell(cell)) {
-    state->set_type_at(cell, state->bottom_type());
+  while (state->stack_size() != 0) {
+    state->push(state->bottom_type());
```

Comment:
It might be more clear if the condition is `< 0` since we only enter the loop if the stack size is negative. Maybe we could also assert that the stack size is `<= 0` before the loop? Otherwise, nice cleanup!

```suggestion
  assert(state->stack_size() <= 0, "stack size should not be positive");
  while (state->stack_size() < 0) {
    state->push(state->bottom_type());
```